regulation-worker/internal/destination/destination.go

```diff
@@ -77,13 +78,12 @@ func (d *DestMiddleware) getDestDetails(ctx context.Context) (backendconfig.Conf
 	bo.MaxInterval = time.Minute
 	bo.MaxElapsedTime = maxWait
 	var destConf backendconfig.ConfigT
-	var ok bool
 	if err = backoff.Retry(func() error {
 		pkgLogger.Debugf("Fetching backend-config...")
 		// TODO : Revisit the Implementation for Regulation Worker in case of MultiTenant Deployment
-		destConf, ok = d.Dest.Get(config.GetWorkspaceToken())
-		if !ok {
-			return fmt.Errorf("error while getting destination details")
+		destConf, err = d.Dest.Get(ctx, config.GetWorkspaceToken())
+		if err != nil {
+			return fmt.Errorf("error while getting destination details: %v", err)
```

Comment:
shouldn't we generally wrap errors?
```suggestion
			return fmt.Errorf("error while getting destination details: %w", err)
```